src/FsCheck/Runner.fs

```diff
@@ -377,19 +380,27 @@ module Runner =
         let tryShrinkNb = ref 0
         let origArgs = ref []
         let lastStep = ref (Failed Res.rejectedV)
-        let seed = match config.Replay with None -> Random.create() | Some s -> s
+        let seed, size = match config.Replay with None -> Random.create(), None | Some s -> s.Rnd, s.Size
         let increaseSizeStep = float (config.EndSize - config.StartSize) / float config.MaxTest
+        let preLastSeed = ref seed
+        let preLastSize = ref (int ((float <| defaultArg size config.StartSize) - increaseSizeStep))
+        let lastSeed = ref seed
+        let lastSize = ref (int (float <| defaultArg size config.StartSize))
```

Comment:
Well, coding at 3 am weekdays never a good idea. If I'd choose to separate `test'` that ugliness wouldn't be here. Will hack another approach today.